src/System.Linq/src/System/Linq/Enumerable.Sum.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        public static int Sum(this IEnumerable<int> source)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            int sum = 0;
+            checked
+            {
+                foreach (int v in source) sum += v;
+            }
+            return sum;
+        }
+
+        public static int? Sum(this IEnumerable<int?> source)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            int sum = 0;
+            checked
+            {
+                foreach (int? v in source)
+                {
+                    if (v != null) sum += v.GetValueOrDefault();
```

Comment:
This applies to all "primitive?" cases. Perhaps it is better to do as a separate change and apply to all the cases. There could be other operators using this pattern, and such... 
It would be easier to concentrate on one kind of change.